### PR TITLE
329 bggaccount

### PR DESCRIPTION
**Pull request by vfportero**
Simple integration with BGG to allow relate NemeStats accountwith BGG user.

Just go to Manage Account view and set your BGG User. 

This is it (for now...). In the future: Mass update board games from BGG Account and Import game plays from BGG :)

### CODE REVIEW

File: Source/BusinessLogic/Logic/Users/BoardGameGeekUserSaver.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            this._dataContext = dataContext;
+        }
+
+        public BoardGameGeekUserDefinition CreateUserDefintion(CreateBoardGameGeekUserDefinitionRequest request, ApplicationUser currentUser)
+        {
+            ValidateRequest(request);
+            ValidateCurrentUserHasBGGAccount(currentUser);
+
+            var existingItem = _dataContext.GetQueryable<BoardGameGeekUserDefinition>().FirstOrDefault(u => u.Id == request.BoardGameGeekUserId);
```
Comment (jakejgordon) [inline Source/BusinessLogic/Logic/Users/BoardGameGeekUserSaver.cs]: You can probably eliminate the separate request to .Any() and just check to see if you get a result here. No need to query the DB twice (one for .Any() and 2nd time for .FirstOrDefault()).

File: Source/BusinessLogic/Logic/Users/UserRetriever.cs [hunk 2]
```diff
@@ -23,27 +24,48 @@ public UserInformation RetrieveUserInformation(string userId, ApplicationUser ap
...
+                    }).ToList(),
+                    Players = user.Players.Select(player => new PlayerInfoForUser
+                    {
+                        PlayerId = player.Id,
+                        PlayerName = player.Name,
+                        GamingGroupId = player.GamingGroupId
+                    }).ToList(),
+                    BoardGameGeekUser = new BoardGameGeekUserInformation
+                    {
+                        Name = user.BoardGameGeekUser.Name,
```
Comment (jakejgordon) [inline Source/BusinessLogic/Logic/Users/UserRetriever.cs]: Does this blow up if they don't have a BoardGameGeekUser?

Comment (vfportero) [inline Source/BusinessLogic/Logic/Users/UserRetriever.cs]: Nop, return null at that case

File: Source/BusinessLogic/Logic/Users/BoardGameGeekUserSaver.cs (hunk 1 above)
Comment (vfportero) [inline Source/BusinessLogic/Logic/Users/BoardGameGeekUserSaver.cs]: changed :)

**Pull request merged**